website/src/pages/DailyPage/components/DailyCharts.tsx

```diff
@@ -0,0 +1,307 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { ChartContainer, ChartTooltip } from "@/components/ui/chart";
+import { Skeleton } from "@/components/ui/skeleton";
+import { MacroData, Workloads } from "@/types";
+import useApiCall from "@/utils/Hook";
+import { secondToMicrosecond } from "@/utils/Utils";
+import { useEffect, useState } from "react";
+import { CartesianGrid, Legend, Line, LineChart, XAxis, YAxis } from "recharts";
+
+export type DailyChartsProps = {
+  benchmarkType: string;
+};
+
+type NumberDataPoint = { x: string; y: number };
+type StringDataPoint = { x: string; y: string };
+
+type ChartDataItem =
+  | { id: string; data: NumberDataPoint[] }
+  | { id: string; data: StringDataPoint[] };
+
+type ChartDatas = {
+  data: ChartDataItem[];
+  title: string;
+  colors: string[];
+};
+
+const chartConfig: { [key: string]: { label: string; color: string } } = {
+  qpsReads: {
+    label: "Reads",
+    color: "hsl(var(--chart-qpsReads))",
+  },
```

Comment:
I am guessing the keys here are what we should change in order to have proper names in the legend?

Reply:
Ah nevermind, I think it is in `chartMetadas`